analysis/src/python/data_analysis/preprocessing/filter_datasets.py

```diff
@@ -0,0 +1,45 @@
+import argparse
+import logging
+import sys
+
+from analysis.src.python.utils.df_utils import filter_df_by_iterable_value, merge_dfs, read_df, write_df
+from analysis.src.python.utils.file_utils import create_directory, get_output_filename
+from analysis.src.python.utils.logging_utils import configure_logger
+
+
+def filer_datasets(first_dataset_path: str, second_dataset_path: str, first_column: str, second_column: str,
```

Comment:
It looks strange...Maybe should we add something about merging? Also, it would be great to accept the list of pairs (dataset_path, column in the dataset) to extend it on any number of arguments. For parsing dict, you can use something like [this](https://stackoverflow.com/questions/29986185/python-argparse-dict-arg) 
What do you think?

Reply:
This file is not used in the analysis actually, I just added it just for checking that when we join users and submissions in python a lot of submissions disappear. So I deleted it as we do not really need it in the analysis process)